packages/components/src/InputTime/InputTime.rebuilt.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { ChangeEvent, useId, useRef } from "react";
+import { useTimePredict } from "./hooks/useTimePredict";
+import { InputTimeProps, InputTimeRebuiltProps } from "./InputTime.Types";
+import { dateToTimeString, timeStringToDate } from "./utils/input-time-utils";
+import { FormFieldWrapper, useFormFieldWrapperStyles } from "../FormField";
+
+export function InputTimeRebuilt({
+  defaultValue,
+  value,
+  onChange,
+  ...params
+}: InputTimeRebuiltProps) {
+  const ref =
+    (params.inputRef as React.RefObject<HTMLInputElement>) ??
+    useRef<HTMLInputElement>(null);
+  const { setTypedTime } = useTimePredict({
+    value,
+    handleChange,
+  });
+
+  const { inputStyle } = useFormFieldWrapperStyles(params);
+
+  const id = getId(params);
+
+  const valueProps = {
+    ...(defaultValue && { defaultValue: dateToTimeString(defaultValue) }),
+    ...(!defaultValue && { value: dateToTimeString(value) }),
+  };
+
+  return (
+    <FormFieldWrapper
+      disabled={params.disabled}
+      size={params.size}
+      align={params.align}
+      inline={params.inline}
+      name={params.name}
+      error={""}
+      identifier={id}
+      descriptionIdentifier={`descriptionUUID--${id}`}
+      invalid={Boolean(params.invalid)}
+      description={params.description}
+      clearable={params.clearable ?? "never"}
```

Comment:
I noticed that clearable doesn't work when only a `defaultValue` is provided and the component is uncontrolled. This is likely due to [this](https://github.com/GetJobber/atlantis/blob/JOB-120387/inputtime-but-without-rhf/packages/components/src/FormField/FormFieldWrapper.tsx#L71-L78) line. I found that if `readonly value: Date | undefined;`  (note that this isn't `readonly value?: Date` is added to  `InputTimeRebuiltProps` the value needs to be passed but it can be undefined. This way the `InputTimeRebuilt` would only be usable as a controlled input.

Reply:
interesting. I am going to try to remove `defaultValue` as a possible prop, so if that's the only scenario where clearable is misbehaving it _should_ go away soon.